adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/android/BrokerProxy.cs

```diff
@@ -454,6 +450,22 @@ private Bundle GetBrokerOptions(AuthenticationRequest request)
             return brokerOptions;
         }
 
+        private void ValidateBrokerRedirectURI(AuthenticationRequest request)
+        {
+            string computedRedirectUri = GetRedirectUriForBroker();
+
+            //During the silent broker flow, the redirect URI will be null.
+            if (string.IsNullOrEmpty(request.RedirectUri))
+            {
```

Comment:
if you're just checking for a null redirectUri (for silent flows), then check before even going into this method. then you're not calculating the computedRedirectUri too and wasting cycles.